components/core/src/clp_s/JsonParser.cpp

```diff
@@ -557,6 +621,307 @@ int32_t JsonParser::add_metadata_field(std::string_view const field_name, NodeTy
+                } else {
+                    SPDLOG_ERROR("Encoded string is not utf8 compliant: \"{}\"", decoded_value);
+                    throw OperationFailed(ErrorCodeFailure, __FILENAME__, __LINE__);
+                }
+                m_current_parsed_message.add_value(node_id, encoded_str);
+            } break;
+            case NodeType::UnstructuredArray: {
+                std::string array_str;
+                if (pair.second.value().is<clp::ir::EightByteEncodedTextAst>()) {
+                    array_str = pair.second.value()
+                                        .get_immutable_view<clp::ir::EightByteEncodedTextAst>()
+                                        .decode_and_unparse()
+                                        .value();
+                } else {
+                    array_str = pair.second.value()
+                                        .get_immutable_view<clp::ir::FourByteEncodedTextAst>()
+                                        .decode_and_unparse()
+                                        .value();
+                }
+                m_current_parsed_message.add_value(node_id, array_str);
+                break;
+            }
+            default:
+                // Don't need to add value for obj or null
+                break;
+        }
+        m_current_schema.insert_ordered(node_id);
+    }
+
+    int32_t const current_schema_id = m_archive_writer->add_schema(m_current_schema);
+    m_current_parsed_message.set_id(current_schema_id);
+    m_archive_writer->append_message(current_schema_id, m_current_schema, m_current_parsed_message);
+}
+
+auto JsonParser::parse_from_ir() -> bool {
+    for (auto& file_path : m_file_paths) {
+        auto const fsize = std::filesystem::file_size(file_path);
+        if (0 == fsize) {
+            m_archive_writer->close();
+            return false;
+        }
+        clp::streaming_compression::zstd::Decompressor decompressor;
+        size_t curr_pos = 0;
+        size_t last_pos = 0;
+        decompressor.open(file_path);
+
+        auto deserializer_result{Deserializer<IrUnitHandler>::create(decompressor, IrUnitHandler{})
+        };
+        if (deserializer_result.has_error()) {
+            decompressor.close();
+            m_archive_writer->close();
+            return false;
+        }
+        auto& deserializer = deserializer_result.value();
+        auto& ir_unit_handler{deserializer.get_ir_unit_handler()};
+
+        int32_t log_event_idx_node_id{};
+        auto add_log_event_idx_node = [&]() {
+            if (m_record_log_order) {
+                log_event_idx_node_id
+                        = add_metadata_field(constants::cLogEventIdxName, NodeType::Integer);
+            }
+        };
+        add_log_event_idx_node();
+        while (true) {
+            auto const kv_log_event_result{deserializer.deserialize_next_ir_unit(decompressor)};
+
+            if (kv_log_event_result.has_error()) {
+                break;
```

Comment:
```suggestion
                m_archive_writer->close();
                decompressor.close();
                return false;
```
Sorry, hadn't noticed this previously, but we should fail ingestion on errors from the IR deserializer.